src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+            0x00, /* 0x75 : distance threshold low LSB */ 0x00, /* 0x76 : not user-modifiable */
+            0x01, /* 0x77 : not user-modifiable */ 0x0f, /* 0x78 : not user-modifiable */
+            0x0d, /* 0x79 : not user-modifiable */ 0x0e, /* 0x7a : not user-modifiable */
+            0x0e, /* 0x7b : not user-modifiable */ 0x00, /* 0x7c : not user-modifiable */
+            0x00, /* 0x7d : not user-modifiable */ 0x02, /* 0x7e : not user-modifiable */
+            0xc7, /* 0x7f : ROI center, use SetROI() */ 0xff, /* 0x80 : XY ROI (X=Width, Y=Height), use SetROI() */
+            0x9B, /* 0x81 : not user-modifiable */ 0x00, /* 0x82 : not user-modifiable */
+            0x00, /* 0x83 : not user-modifiable */ 0x00, /* 0x84 : not user-modifiable */
+            0x01, /* 0x85 : not user-modifiable */ 0x00, /* 0x86 : clear interrupt, use ClearInterrupt() */
+            0x00 /* 0x87 : start ranging, use StartRanging() or StopRanging(), If you want an automatic start after VL53L1X_init() call, put 0x40 in location 0x87 */
+        };
+
+        private I2cDevice _i2CDevice = null!;
+        private bool _rangingInitialized;
+        private GpioController? _controller;
+
+        /// <summary>
+        /// Creates a Vl53L1X sensor class
+        /// </summary>
+        /// <param name="i2cAddress">The I2C address for the device</param>
+        /// <param name="busId">The id of the I2C bus</param>
+        /// <param name="xShutPin">
+        /// The xShut pin number used to power up the device.
+        /// If omitted, The xShutPin must be tied to the power supply value through a pull up resistor
+        /// </param>
+        /// <param name="operationTimeoutMilliseconds">Timeout for reading data, by default 500 milliseconds</param>
+        public Vl53L1X(byte i2cAddress = (byte)Registers.VL53L1X_DEFAULT_DEVICE_ADDRESS, int busId = 1, int xShutPin = -1, int operationTimeoutMilliseconds = 500)
+        {
+            _xShutPin = xShutPin;
+            _operationTimeoutMilliseconds = operationTimeoutMilliseconds;
+
+            PowerOn();
+            ChangeI2CAddress(i2cAddress, busId);
+            WaitForBooted();
+            InitSensor();
+        }
+
+        /// <summary>
+        ///     Get a distance in millimeters.
+        ///     If ranging has not been started yet, the function will automatically start the ranging feature of the device.
+        /// </summary>
+        public ushort Distance
+        {
+            get
+            {
+                if (!_rangingInitialized)
+                {
+                    StartRanging();
+                }
+
+                WaitForDataReady();
+                ushort distance = GetDistance();
+                return distance;
+            }
+        }
+
+        private void PowerOn()
+        {
+            if (_xShutPin < 0)
+            {
+                return;
+            }
+
+            _controller = new GpioController();
```

Comment:
this is not allowed neither. Please pass a GpioController in the constructor, default null with the shouldDispose pattern. And only in the constructor, you can create if passed null. See here: https://github.com/dotnet/iot/blob/b1cee19c6223a8755bd6fabff41f2f6332ced73c/src/devices/Button/GpioButton.cs#L32

Reply:
I removed the gpiocontroller completely from the class